otj-kafka-spring/src/main/java/com/opentable/kafka/spring/builders/KafkaConsumerFactoryBuilder.java

```diff
@@ -0,0 +1,74 @@
+package com.opentable.kafka.spring.builders;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.listener.ContainerProperties.AckMode;
+import org.springframework.lang.Nullable;
+
+import com.opentable.kafka.builders.EnvironmentProvider;
+import com.opentable.kafka.builders.KafkaConsumerBaseBuilder;
+
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class KafkaConsumerFactoryBuilder<K, V>  extends KafkaConsumerBaseBuilder<KafkaConsumerFactoryBuilder<K, V>, K, V> {
+
+    private Optional<Duration> pollTimeout = Optional.empty();
+    private Optional<AckMode> ackMode = Optional.empty();
+
+
+    KafkaConsumerFactoryBuilder(Map<String, Object> props, EnvironmentProvider environmentProvider) {
+        super(props, environmentProvider);
+    }
+
+    @Override
+    protected KafkaConsumerFactoryBuilder<K, V> self() {
+        return this;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <K2, V2> KafkaConsumerFactoryBuilder<K2, V2> withDeserializers(Class<? extends Deserializer<K2>> keyDeSer, Class<? extends Deserializer<V2>> valDeSer) {
+        return (KafkaConsumerFactoryBuilder<K2, V2>) super.withDeserializers(keyDeSer, valDeSer);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <K2, V2> KafkaConsumerFactoryBuilder<K2, V2> withDeserializers(Deserializer<K2> keyDeSer, Deserializer<V2> valDeSer) {
+        return (KafkaConsumerFactoryBuilder<K2, V2>) super.withDeserializers(keyDeSer, valDeSer);
+    }
+
+    public KafkaConsumerFactoryBuilder<K, V> withPollTimeout(Duration pollTimeout) {
```

Comment:
Are all the rest inherited?

Reply:
why is withPollTimeout and withAckMode here if it's inherited?

Reply:
Hmm.. Because those are `ContainerProperties` https://docs.spring.io/spring-kafka/api/org/springframework/kafka/listener/ContainerProperties.html#setPollTimeout-long- Not Kafka ones.